vault/plugincatalog/plugin_catalog.go

```diff
@@ -676,8 +681,13 @@ func (c *PluginCatalog) getDatabaseRunningVersion(ctx context.Context, pluginRun
 
 		return logical.EmptyPluginVersion, nil
 	}
-	merr = multierror.Append(merr, fmt.Errorf("failed to load plugin as database v4: %w", err))
-	return logical.EmptyPluginVersion, merr
+
+	merr = multierror.Append(merr, err)
+	if pluginRunner.OCIImage != "" {
+		return logical.EmptyPluginVersion, fmt.Errorf("%w: %s", ErrContainerizedPluginUnableToRun, merr)
```

Comment:
nit: Same as above if you agree.

```suggestion
		return logical.EmptyPluginVersion, fmt.Errorf("%w: %w", ErrContainerizedPluginUnableToRun, merr.ErrorOrNil())
```